R/forecast.R

```diff
@@ -426,6 +432,33 @@ new_forecast <- function(data, classname) {
 }
 
 
+#' Remakes a forecast object with a subclass
+#'
+#' This function takes a forecast object and modifies its class name by removing
+#' the old class name and adding a new class name. It also performs an assertion
+#' check on the modified forecast object.
+#'
+#' @param forecast The forecast object to be modified
+#' @param old_classname The name of the class to be removed from the forecast
+#' object
+#' @param new_classname The name of the class to be added to the forecast object
+#' @param verbose A logical value indicating whether to print verbose output
+#'
+#' @return The modified forecast object
+#' @keywords internal
+remake_forecast <- function(
```

Comment:
Alternatively, we could just call `as_forecast()` instead. The current `sample_to_quantile()` function does that. 
I'd be happy to avoid the additional code complexity of introducing a new function.

Reply:
I tried to do this hence the new checking option and it still wouldn't work. This what motivated the question about refactoring `as_forecast` as the current approach with the type specific checking really doesn't seem feasible for future extensions. 

I think we should make a new issue for this (related to the `as_forecast` issue). There is also a more general point about whether or not `as_forecast` should be able to reclass a `forecast` object or whether it does actually make sense workflow wise to have something that explicitly removes and then readds the class.